depends/thirdparty/orc/tools/src/HdfsFileScan.cc

```diff
@@ -0,0 +1,65 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <string>
+#include <memory>
+#include <iostream>
+#include <string>
+
+#include "orc/ColumnPrinter.hh"
+
+#include "Exceptions.hh"
+
+#include "wrap/libhdfs3-wrapper.h"
+
+int main(int argc, char* argv[]) {
+  if (argc < 2) {
+    std::cout << "Usage: orc-scan <filename>\n";
```

Comment:
Good advice! We will use prefix to auto-detect the protocol. For default, we regard it as `file://`.